tensorboard/webapp/notification_center/_redux/testing.ts

```diff
@@ -33,3 +42,37 @@ export function buildStateFromNotificationState(
 ): State {
   return {[NOTIFICATION_FEATURE_KEY]: runsState};
 }
+
+@Injectable()
+export class TestingNotificationCenterDataSource
+  implements NotificationCenterDataSource {
+  fetchNotification() {
+    return of({
+      response: {notifications: [], error: ''},
+    });
+  }
+}
+
+export function provideTestingNotificationCenterDataSource() {
+  return [
+    TestingNotificationCenterDataSource,
+    {
+      provide: NotificationCenterDataSource,
+      useExisting: TestingNotificationCenterDataSource,
+    },
+  ];
+}
+
+export function buildDataSourceNotification(): DataSourceNotifications {
+  return {
+    notifications: [
+      {
+        category: CategoryEnum.WHATS_NEW,
+        dateInMs: 0,
```

Comment:
Optional, but would recommend using test data that's less likely to appear by chance - using "0" for the date could very well be the actual default value, which wouldn't make this as good of a test, and using the same value for `title` and `content` means that even if we swapped them we wouldn't notice.  So if it were me I'd use something like `123` and `test title` and `random content`.

Reply:
thanks for the example. adjusted